src/main/java/carpet/script/api/Monitoring.java

```diff
@@ -13,62 +13,50 @@
 import carpet.utils.SpawnReporter;
 import it.unimi.dsi.fastutil.objects.Object2IntMap;
 import net.minecraft.entity.SpawnGroup;
-import net.minecraft.server.world.ServerWorld;
 import net.minecraft.world.SpawnHelper;
 
 import java.util.HashMap;
 import java.util.Locale;
 import java.util.Map;
+import java.util.Optional;
 
 public class Monitoring {
+    @ScarpetFunction
+    public static Value system_info(Context c, String property) {
+        Value res = SystemInfo.get(property, (CarpetContext) c);
+        if (res == null) throw new InternalExpressionException("Unknown option for 'system_info': '" + property + "'");
+        return res;
+    }
 
-    public static void apply(Expression expression)
-    {
-        expression.addContextFunction("system_info", -1, (c, t, lv) ->
+    @ScarpetFunction(maxParams = 1)
+    public static Value get_mob_counts(Context c, Optional<String> category) {
+        SpawnHelper.Info info = ((CarpetContext)c).s.getWorld().getChunkManager().getSpawnInfo();
+        if (info == null) return null;
+        Object2IntMap<SpawnGroup> mobcounts = info.getGroupToCount();
+        int chunks = info.getSpawningChunkCount();
+        if (category.isPresent())
         {
-            if (lv.size() == 0)
-            {
-                return SystemInfo.getAll((CarpetContext) c);
-            }
-            if (lv.size() == 1) {
-                String what = lv.get(0).getString();
-                Value res = SystemInfo.get(what, (CarpetContext) c);
-                if (res == null) throw new InternalExpressionException("Unknown option for 'system_info': " + what);
-                return res;
-            }
-            throw new InternalExpressionException("'system_info' requires one or no parameters");
-        });
-        // game processed snooper functions
-        expression.addContextFunction("get_mob_counts", -1, (c, t, lv) ->
+            SpawnGroup group = SpawnGroup.byName(category.get().toLowerCase(Locale.ROOT));
+            if (group == null) throw new InternalExpressionException("Unreconized mob category: " + category.get());
```

Comment:
```suggestion
            if (group == null) throw new InternalExpressionException("Unrecognized mob category: " + category.get());
```

Reply:
Fixed, weird it wasn't noticed yet, since it was in the original implementation.